shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -70,6 +72,75 @@ - (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent*)event {
   return NO;
 }
 
+// Creates and initializes a UIVisualEffectView with a UIBlurEffect. Extracts and returns its
+// gaussianFilter. Logs errors and returns if Apple's API has changed and the filter can't be
+// extracted.
+- (NSObject*)extractGaussianFilter {
+  UIVisualEffectView* visualEffectView = [[UIVisualEffectView alloc]
+      initWithEffect:[UIBlurEffect effectWithStyle:UIBlurEffectStyleLight]];
+
+  UIView* view = [visualEffectView.subviews firstObject];
+  if (!view || ![view isKindOfClass:NSClassFromString(@"_UIVisualEffectBackdropView")]) {
+    FML_DLOG(ERROR) << "Apple's API for UIVisualEffectView changed. Update the implementation to "
+                       "access its subviews.";
+    return nil;
+  }
+
+  NSObject* gaussianFilter = [[view.layer.filters firstObject] retain];
+  if (!gaussianFilter || ![[gaussianFilter valueForKey:@"name"] isEqual:@"gaussianBlur"]) {
+    FML_DLOG(ERROR) << "Apple's API for UIVisualEffectView changed. Update the implementation to "
+                       "access the Gaussian blur filter. ";
+    return nil;
+  }
+
+  [visualEffectView release];
+
+  if (![[gaussianFilter valueForKey:@"inputRadius"]
+          isKindOfClass:[NSNumber class]]) {  // TODO EMILY: is there another way to check that
+                                              // inputRadius key is valid? -> DOCUMENT ATTEMPTS
+    FML_DLOG(ERROR) << "Apple's API for UIVisualEffectView changed. Update the implementation "
+                       "access the Gaussian blur filter's properties.";
+    return nil;
+  }
+
+  return gaussianFilter;
+}
+
+- (void)applyBackdropFilters:(NSArray*)blurRadii {
+  NSObject* gaussianFilter;
```

Comment:
can move this to line 114, right before its value is initialized.

Reply:
It's hidden, but I use gaussianFilter later in the method when adding copies of gaussianBlur to _gaussianBlur. 

I'm adding comments to make it more clear, but in this commit I only instantiated the gaussianFilter if ([blurRadii count] > [_gaussianFilters count]) because that is when more filters need to be added to _gaussianFilters. Then, I copy gaussianFilter in the for loop when it is needed. I wrote it this way to limit the number of calls to create a UIVisualEffectView. 

In my next commit, I save _gaussianFilter as a private instance variable so only one UIVisualEffectView is created. I also add the [_gaussianFilter copy] to _gaussianFilters in a separate loop from the loop where the filters are edited.